plugin/src/test/java/com/flit/protoc/gen/server/TypeMapperTest.java

```diff
@@ -0,0 +1,207 @@
+package com.flit.protoc.gen.server;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import com.google.protobuf.DescriptorProtos;
+import com.squareup.javapoet.ClassName;
+import org.junit.Test;
+
+public class TypeMapperTest {
+
+  private static final String PROTO_PACKAGE = "flit.test";
+  private static final String JAVA_PACKAGE = "com.flit.test";
+
+  private static final DescriptorProtos.DescriptorProto MAP_MESSAGE = DescriptorProtos.DescriptorProto
+      .newBuilder()
+      .setName("Map")
+      .build();
+
+  private static final DescriptorProtos.DescriptorProto MAPPER_MESSAGE = DescriptorProtos.DescriptorProto
+      .newBuilder()
+      .setName("Mapper")
+      .build();
+
+  @Test
+  public void protoPackage() {
+    DescriptorProtos.FileOptions options = DescriptorProtos.FileOptions.newBuilder()
+        .setJavaMultipleFiles(false)
+        .setJavaOuterClassname("Mapper")
+        .build();
+
+    DescriptorProtos.FileDescriptorProto proto = DescriptorProtos.FileDescriptorProto.newBuilder()
+        .setPackage(PROTO_PACKAGE)
+        .setName("mapper.proto")
+        .setOptions(options)
+        .addMessageType(MAP_MESSAGE)
+        .build();
+
+    TypeMapper mapper = new TypeMapper();
+    mapper.add(proto);
+
+    ClassName result = mapper.get(".flit.test.Map");
+    assertEquals(PROTO_PACKAGE, result.packageName());
+    assertEquals("Mapper", result.enclosingClassName().simpleName());
+    assertEquals("Map", result.simpleName());
+  }
+
+  @Test
+  public void protoPackageNameCollision() {
+    DescriptorProtos.FileOptions options = DescriptorProtos.FileOptions.newBuilder()
+        .setJavaMultipleFiles(false)
+        .build();
+
+    DescriptorProtos.FileDescriptorProto proto = DescriptorProtos.FileDescriptorProto.newBuilder()
+        .setPackage(PROTO_PACKAGE)
+        .setName("mapper.proto")
+        .setOptions(options)
+        .addMessageType(MAPPER_MESSAGE)
+        .build();
+
+    TypeMapper mapper = new TypeMapper();
+    mapper.add(proto);
+
+    ClassName result = mapper.get(".flit.test.Mapper");
+    assertEquals(PROTO_PACKAGE, result.packageName());
+    assertEquals("MapperOuterClass", result.enclosingClassName().simpleName());
+    assertEquals("Mapper", result.simpleName());
+  }
+
+  @Test
+  public void protoPackageWithOuterClass() {
```

Comment:
This test is identical to `protoPackage`, if that's the intention, could we remove one of them? If not I thimk for `protoPackage` test we need to remove ` .setJavaOuterClassname("Mapper")`

Reply:
Good catch.  I have gone ahead and removed it as you suggested.